test/extensions/filters/network/common/fuzz/uber_filter.cc

```diff
@@ -0,0 +1,194 @@
+#include "test/extensions/filters/network/common/fuzz/uber_filter.h"
+
+#include "envoy/extensions/filters/network/direct_response/v3/config.pb.h"
+#include "envoy/extensions/filters/network/local_ratelimit/v3/local_rate_limit.pb.h"
+
+#include "common/config/utility.h"
+#include "common/config/version_converter.h"
+#include "common/network/utility.h"
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/network/common/utility.h"
+#include "extensions/filters/network/well_known_names.h"
+
+#include "test/extensions/filters/common/ext_authz/test_common.h"
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+
+std::vector<absl::string_view> UberFilterFuzzer::filterNames() {
+  // This filters that have already been covered by this fuzzer.
+  // Will extend to cover other network filters one by one.
+  static ::std::vector<absl::string_view> filter_names_;
+  if (filter_names_.empty()) {
+    filter_names_ = {NetworkFilterNames::get().ExtAuthorization,
+                     NetworkFilterNames::get().LocalRateLimit,
+                     NetworkFilterNames::get().RedisProxy,
+                     NetworkFilterNames::get().ClientSslAuth,
+                     NetworkFilterNames::get().Echo,
+                     NetworkFilterNames::get().DirectResponse,
+                     NetworkFilterNames::get().DubboProxy,
+                     NetworkFilterNames::get().SniCluster};
+  }
+  return filter_names_;
+}
+
+void UberFilterFuzzer::reset() {
+  // Reset some changes made by current filter on some mock objects.
+
+  // Close the connection to make sure the filter's callback is set to nullptr.
+  read_filter_callbacks_->connection_.raiseEvent(Network::ConnectionEvent::LocalClose);
+  // Clear the filter's raw pointer stored inside the connection_ and reset the connection_.
+  read_filter_callbacks_->connection_.callbacks_.clear();
+  read_filter_callbacks_->connection_.bytes_sent_callbacks_.clear();
+  read_filter_callbacks_->connection_.state_ = Network::Connection::State::Open;
+}
+void UberFilterFuzzer::perFilterSetup(const std::string& filter_name) {
+  // Set up response for ext_authz filter
+  if (filter_name == NetworkFilterNames::get().ExtAuthorization) {
+
+    async_client_factory_ = std::make_unique<Grpc::MockAsyncClientFactory>();
+    async_client_ = std::make_unique<Grpc::MockAsyncClient>();
+
+    ON_CALL(*async_client_, sendRaw(_, _, _, _, _, _))
+        .WillByDefault(testing::WithArgs<3>(Invoke([&](Grpc::RawAsyncRequestCallbacks& callbacks) {
+          Filters::Common::ExtAuthz::GrpcClientImpl* grpc_client_impl =
+              dynamic_cast<Filters::Common::ExtAuthz::GrpcClientImpl*>(&callbacks);
+          const std::string empty_body{};
+          const auto expected_headers =
+              Filters::Common::ExtAuthz::TestCommon::makeHeaderValueOption({});
+          auto check_response = Filters::Common::ExtAuthz::TestCommon::makeCheckResponse(
```

Comment:
might it be possible to fuzz an OK response vs an error or denied response?
also OK with adding as TODO

Reply:
I talked with Sam about this. Since the generic fuzzer should test on generic behaviors, it is a little inefficient to add a new  `action` to choose the response just for ext_authz. Anyway, it is possible to choose the response. I will add a TODO now and if there is a better way to do this I will modify it.

Reply:
The problem with mocking RPC responses is that each OK response will include a payload that is specific to that RPC.  But in this case, you are mocking the specific response for the auth RPC.  So you could mock both success and failure.

I do think we eventually want to return success or failure here that is a function of the bytes passed to the fuzzer.  But I would do that in a separate CL.